cni/pkg/ambient/net_linux.go

```diff
@@ -237,6 +237,49 @@ func getVethWithDestinationOf(ip string) (*netlink.Veth, error) {
 	return veth, nil
 }
 
+func findVethLinkForPeerIP(peerIP net.IP) (*netlink.Veth, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list links: %v", err)
+	}
+	var v netlink.Veth
+	for _, l := range links {
+		if l.Type() != v.Type() {
+			continue
+		}
+		veth := l.(*netlink.Veth)
+		peerIndex, err := getPeerIndex(veth)
+		if err != nil {
+			log.Warnf("failed to get peer index for veth interface %s: %v", veth.Name, err)
+		}
+		peerNs, err := getNsNameFromNsID(veth.Attrs().NetNsID)
+		if err != nil {
+			log.Warnf("failed to get network namespace name from namespace id '%d' for veth interface %s: %v", veth.Attrs().NetNsID, veth.Name, err)
+		}
+		var peerVethFound bool
+		if err := netns.WithNetNSPath(fmt.Sprintf("/var/run/netns/%s", filepath.Base(peerNs)), func(netns.NetNS) error {
+			peerLink, err := netlink.LinkByIndex(peerIndex)
+			if err != nil {
+				return fmt.Errorf("failed to get veth interface '%s' by peer index: %v", veth.Name, err)
+			}
+			addrs, err := netlink.AddrList(peerLink, netlink.FAMILY_V4)
+			if err != nil {
+				return fmt.Errorf("failed to get address for veth interface '%s': %v", veth.Name, err)
+			}
+			if addrs[0].IP.Equal(peerIP) {
+				peerVethFound = true
+			}
```

Comment:
This won't work if `peerIP` happens to be v6 or 4-in-6.

Granted, we have other spots in this code where we make the (invalid) assumptions that 

- `Status.PodIP` is the only IP a pod has
- `Status.PodIP` is a V4 address

so this is mostly a nit for this PR.